### PR TITLE
feat: Feature gate session replay controls using available_product_features

### PR DESCRIPTION
**Pull request by xrdt**
## Problem

Need to only allow new users to have access to these controls. 

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/types.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ import { NodeKind } from './queries/schema'
 
 export type Optional<T, K extends string | number | symbol> = Omit<T, K> & { [K in keyof T]?: T[K] }
 
-// Keep this in sync with backend constants (constants.py)
+// Keep this in sync with backend constants/features/{product_name}.yml
```
Comment (xrdt) [inline frontend/src/types.ts:44]: Feels like another thing we could automatically publish from billing so we don't have to manually keep it in sync.

**Review by benjackwhite — CHANGES_REQUESTED** [review]
Perhaps I've missed something but this doesn't feel like something we should control with billing features...

Why would we only offer it if you are subscribed? Even on a free tier it makes a lot of sense and keeps people using it. If you are thinking of upgrading but are worried about cost, maybe the thing you would do first is add the controls before upgrading.
Similarly if you downgrade but had some testings configured, you're now stuck with whatever setting you had...

Unless there is really strong reasoning why we would do this, I would personally vote to keep it as a flagged feature and gradually roll out to everyone. If we want to limit to only new users - feature flag, keeps it simple and allows us to conditionally add more legacy users for example as a tool for customer success.

**Comment by raquelmsmith** [comment]
@benjackwhite with the new plans it won't only be offered when you are sub'd. It'll be available on the free plans too.

BUT because we want to pair the rollout of more sensible pricing _with_ the rollout of the controls (ie. you get these controls, but you also have a lower free tier to accommodate that), that means that existing plans (with the higher free tier) won't get access to the controls.

We know what plans they are on from billing, so adding the feature gating via billing makes a lot of sense.

Once we see how much people use these and what the impacts are we will consider migrating everyone to the new plans - until then if people want them they can be put on the new plans manually.

**Review by raquelmsmith — CHANGES_REQUESTED** [review]

**Comment by xrdt** [comment]
You can now control whether the settings show either with the feature flag or by setting individual settings: 

https://github.com/PostHog/posthog/assets/21014901/a3ea21d7-09fc-46f8-9399-48a29ef3ae4e

**Review by raquelmsmith — APPROVED** [review]

**Comment by raquelmsmith** [comment]
This can go in now, no need to wait for billing launch since we are still looking for the flag value.

**Review by benjackwhite — APPROVED** [review]
Happy to defer to your judgement. Still feels weird that we would need both available features _and_ feature flag but if it works, it works.

**Review by pauldambra — APPROVED** [review]
There aren't many people with the flag on, so we can follow up by checking price impact of moving them to the new plans to get to a place where we can remove the flag but there's no rush for that

**Pull request merged**